### PR TITLE
Fix Graph break from `Tensor.item()`

### PR DESCRIPTION
**Pull request by ValeGian**
## Motivation

This PR aims to fix a `Graph break from Tensor.item()` issue when training Eagle3, additionally obtaining a slight speedup.

While training I got
```
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]     torch._dynamo.config.capture_scalar_outputs = True
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0] or:
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]     env TORCHDYNAMO_CAPTURE_SCALAR_OUTPUTS=1
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0] to include these operations in the captured graph.
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0] Graph break: from user code at:
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]   File "/home/ubuntu/SpecForge/.venv/lib/python3.12/site-packages/specforge/core/eagle3.py", line 789, in _compute_metric_acc
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]     ).sum().item() / (loss_mask.sum().item() + 1e-6)
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]
[rank1]:W0906 20:38:55.305000 5696 torch/_dynamo/variables/tensor.py:1047] [7/0]
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0] Graph break from `Tensor.item()`, consider setting:
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0]     torch._dynamo.config.capture_scalar_outputs = True
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0] or:
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0]     env TORCHDYNAMO_CAPTURE_SCALAR_OUTPUTS=1
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0] to include these operations in the captured graph.
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0]
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0] Graph break: from user code at:
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0]   File "/home/ubuntu/SpecForge/.venv/lib/python3.12/site-packages/specforge/core/eagle3.py", line 789, in _compute_metric_acc
[rank0]:W0906 20:38:55.305000 5695 torch/_dynamo/variables/tensor.py:1047] [7/0]     ).sum().item() / (loss_mask.sum().item() + 1e-6)
```
Any operation that converts a tensor to a Python scalar forces a graph break. The suggested fix is to enable scalar-capture, but that's sub-optimal.

## Modifications

- Removed `.item()` calls from `_compute_metric_acc`
- replaced `+ 1e-6` with a `.clamp_min(1e-6)`
- every call to the model is now followed by a `torch.stack(acces).cpu().tolist()` outside of compiled regions

## Related Issues

NA

## Accuracy Test

`python -m unittest discover -s ./tests -p "test_*.py"` runs successfully

In the following files you'll find 2 training runs of the `train_eagle3_onine.py` over a `Llama3 8B` model with 100k data samples.
[eagle3-training-before-fix.txt](https://github.com/user-attachments/files/22207236/eagle3-old-training.txt)
[eagle3-training-after-fix.txt](https://github.com/user-attachments/files/22207235/eagle3-new-training.txt)
Accuracies and losses at the end of the training are exactly the same.

## Benchmark & Profiling

As can be seen in the previously attached training logs, the toy training before the fix runs in `14:43 minutes` while the training after the fix runs in `14:18 minutes`, denoting a slight speedup. This remained consistent even after relaunching the training.

## Checklist

- [x] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [x] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

File: scripts/train_eagle3_offline.py [hunk 1]
```diff
@@ -417,6 +417,7 @@ def main():
                 hidden_states=data["hidden_state"].cuda(),  # [B, S, D]
                 target=data["target"].cuda(),  # [B, S, D*3]
             )
+            acces = [x.item() for x in acces]
```
[the remark this answers is not in the document]
Comment (sleepcoo) [reply scripts/train_eagle3_offline.py]: gemini is right

[the remark this answers is not in the document]
Comment (ValeGian) [reply scripts/train_eagle3_offline.py]: @sleepcoo Done with https://github.com/sgl-project/SpecForge/pull/222/commits/9ac410d4cd0f12c3764ed4e161d3923df1c7297b

File: scripts/train_eagle3_online.py [hunk 2]
```diff
@@ -518,6 +518,7 @@ def main():
                     attention_mask=data["attention_mask"].cuda(),
                     loss_mask=data["loss_mask"].cuda(),
                 )
+            acces = [x.item() for x in acces]
```
[the remark this answers is not in the document]
Comment (ValeGian) [reply scripts/train_eagle3_online.py]: Done with https://github.com/sgl-project/SpecForge/pull/222/commits/9ac410d4cd0f12c3764ed4e161d3923df1c7297b

**Comment by zyksir** [comment]
Looks good to me! Thank you @ValeGian  Let me merge this after it passes the CI check
